thetis/equation.py

```diff
@@ -62,6 +65,10 @@ class Equation(object):
     SUPPORTED_LABELS = ['source', 'explicit', 'implicit', 'nonlinear']
```

Comment:
It seems `nonlinear` is not actually used anywhere... Could potentially remove that.